geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/functions/RegionCreateFunction.java

```diff
@@ -99,10 +99,18 @@ public void execute(FunctionContext context) {
     } catch (IllegalArgumentException e) {
       resultSender.lastResult(handleException(memberNameOrId, e.getMessage(), e));
     } catch (RegionExistsException e) {
-      String exceptionMsg =
-          CliStrings.format(CliStrings.CREATE_REGION__MSG__REGION_PATH_0_ALREADY_EXISTS_ON_1,
-              regionCreateArgs.getRegionPath(), memberNameOrId);
-      resultSender.lastResult(handleException(memberNameOrId, exceptionMsg, e));
+      if (regionCreateArgs.isIfNotExists()) {
```

Comment:
Remove the duplicate code block that starts at line 69. Instead of checking before creating the region that it exists and then also handling it here; just handle it here (in the new exception catch).

Also add a unit test for this class that tests this new code path.

Reply:
Thanks for the suggestion. I've done that and added a unit test.